packages/dom/src/dom/get-scroll-container.js

```diff
@@ -7,22 +7,37 @@ import getComputedStyle from './get-computed-style';
  * Given a DOM node, finds the closest scrollable container node or the node
  * itself, if scrollable.
  *
- * @param {Element | null} node Node from which to start.
- *
+ * @param {Element | null} node      Node from which to start.
+ * @param {?string}        direction Direction of scrollable container to search for ('vertical', 'horizontal', 'all').
+ *                                   Defaults to 'vertical'.
  * @return {Element | undefined} Scrollable container node, if found.
  */
-export default function getScrollContainer( node ) {
+export default function getScrollContainer( node, direction = 'vertical' ) {
 	if ( ! node ) {
 		return undefined;
 	}
 
-	// Scrollable if scrollable height exceeds displayed...
-	if ( node.scrollHeight > node.clientHeight ) {
-		// ...except when overflow is defined to be hidden or visible
-		const { overflowY } = getComputedStyle( node );
+	if ( direction === 'vertical' || direction === 'all' ) {
+		// Scrollable if scrollable height exceeds displayed...
+		if ( node.scrollHeight > node.clientHeight ) {
+			// ...except when overflow is defined to be hidden or visible
+			const { overflowY } = getComputedStyle( node );
+
+			if ( /(auto|scroll)/.test( overflowY ) ) {
+				return node;
```

Comment:
If "direction" is "all", won't we need to check if the closest horizontal container isn't closer than the closest vertical before returning?

Reply:
Good question, but I'm not sure if I understand the problem. If we're searching by `all`, I didn't think it'd matter which it encounters first, as it should return the first scrollable container of any kind of direction. So if it doesn't match against `vertical`, then the `horizontal` block below will be run and it'll return the node as expected.

Can you think of a case where that logic would slip up?

Reply:
Update: I think I understand the question now. `getScrollContainer` works iteratively only through a single level at a time, recursing via calling itself with `node.parentNode`, so we should be safe to check vertical then check horizontal, as in each iteration we're only looking one level up, rather than looking all the way up in each vertical/horizontal code block.

Reply:
Oh got it, that should work then!